src/main/java/com/imsweb/datagenerator/naaccr/rule/tumor/SiteRule.java

```diff
@@ -21,12 +24,23 @@ public SiteRule() {
     }
 
     @Override
-    public void execute(Map<String, String> tumor, List<Map<String, String>> otherTumors, NaaccrDataGeneratorOptions options) {
-
-        SiteFrequencyDto dto = DistributionUtils.getSite(tumor.get("sex"));
-        tumor.put("primarySite", dto.getSite());
-        tumor.put("histologyIcdO3", dto.getHistology());
-        tumor.put("behaviorIcdO3", dto.getBehavior());
+    public void execute(Map<String, String> tumor, List<Map<String, String>> otherTumors, NaaccrDataGeneratorOptions options, Map<String, Object> context) {
+
+        if (context.get(CONTEXT_FLAG_CURRENT_TUMOR_INDEX) != null) {
```

Comment:
Even if it's "cheap" in this case, it's bad practice to get a value from a map to check if it's null, and then right after, getting it again to use it in some logic.

This can be re-written this way:
```java
Integer currentTumorIndex = (Integer)context.get(CONTEXT_FLAG_CURRENT_TUMOR_INDEX);
Map<Integer, SiteFrequencyDto> siteFreqMap = (Map<Integer, SiteFrequencyDto>)context.get(CONTEXT_FLAG_SITE_FREQ_MAP);
if (currentTumorIndex != null && siteFreqMap != null) {
    ....
}
```